requests/models.py

```diff
@@ -343,6 +343,9 @@ def prepare_url(self, url, params):
         else:
             url = unicode(url) if is_py2 else str(url)
 
+        # Remove leading and trailing whitespaces from url
+        url = url.strip()
```

Comment:
So while I agree that we want to do this, I think we should be cautious. Stripping trailing whitespace will be a behavioural change because right now trailing whitespace *does* work (albeit in a way that is a bit unexpected to people).

So, let's just do `lstrip()` here: we only want to remove leading whitespace.